processor/src/main/java/org/mapstruct/ap/internal/model/VirtualField.java

```diff
@@ -0,0 +1,42 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.internal.model;
+
+import org.mapstruct.ap.internal.model.source.builtin.BuildInFieldReference;
+
+/**
+ * A mapper instance field, initialized as null
+ *
+ * @author Sjaak Derksen
+ */
+public class VirtualField extends Field {
```

Comment:
Why is it "virtual"?

Reply:
Because of the same reason the method that uses the field is called `Virtual`. :smile: . The `BuiltinMethod` and `HelperMethod` are wrapped in a `VirtualMappingMethod ` that binds the template to a method. I've no idea who came up with the name `Virtual`, but the blame perhaps offers some clues :smile:

Reply:
renamed it to `SupportingMappingMethod`